lib/hooks/api/useGetStarSearchWorkspaceHistory.ts

```diff
@@ -0,0 +1,92 @@
+import useSWR, { Fetcher } from "swr";
+import { useEffect, useState } from "react";
+import { publicApiFetcher } from "lib/utils/public-api-fetcher";
+
+export interface StarSearchHistoryItem {
+  id: string;
+  created_at: string;
+  updated_at: string;
+  archived_at: string | null;
+  deleted_at: string | null;
+  public_link: string | null;
+  is_publicly_viewable: boolean;
+  title: string;
+  thread_summary: string;
+}
+
+interface PaginatedResponse {
+  readonly data: StarSearchHistoryItem[];
+  readonly meta: Meta;
+}
+
+type Mutation = ReturnType<typeof useSWR<PaginatedResponse, Error>>["mutate"];
+
+export const useGetStarSearchWorkspaceHistory = ({
+  workspaceId,
+  limit = 30,
+}: {
+  workspaceId: string | undefined;
+  limit?: number;
+}) => {
+  // Contains the mutate functions for each page of data
```

Comment:
We're accumulating the mutations and keeing the thread history in component state for now because we're concatenating the history via a <kbd>load more...</kbd> button in the UI.

In another iteration, a virtualize list will probably make sense, but we may need to manage the mutations like this as well.